src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -160,28 +161,39 @@ private void CheckInvariants()
         var projectId = newProjectState.Id;
 
         var newDependencyGraph = newSolutionState.GetProjectDependencyGraph();
-        var newTrackerMap = CreateCompilationTrackerMap(projectId, newDependencyGraph);
+        var newTrackerMap = CreateCompilationTrackerMap(projectId, newDependencyGraph, modifyTrackerMapInfo, (translate, forkTracker, projectId, newProjectState));
```

Comment:
*nit*.  *consider* inlining.  I've found it easier to read/reason about when you just have a static lambda that takes some data, and an args parameter, and then you see the args passed in after.  BUt this is up to you.

Reply:
inlined, I find that easier to read too, I was just following what was done with CanReuse in the file